database/migrations/2019_12_27_145604_add_storm_league_enum.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\Facades\Schema;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Database\Migrations\Migration;
+
+class AddStormLeagueEnum extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        DB::statement('ALTER TABLE `hotsapi`.`replays` CHANGE COLUMN `game_type` `game_type` ENUM(\'QuickMatch\', \'UnrankedDraft\', \'HeroLeague\', \'TeamLeague\', \'Brawl\', \'StormLeague\') NULL DEFAULT NULL');
```

Comment:
I think we should avoid using raw statements in migrations. See https://github.com/hotsapi/hotsapi/blob/master/database/migrations/2017_08_16_061032_create_replays_table.php#L21 for an example of how to define enum

Reply:
I did try and do it in a database-agnostic way but I got the following error:

```
[Doctrine\DBAL\DBALException]
Unknown column type "enum" requested. Any Doctrine type that you use has to be registered with \Doctrine\DBAL\Types\Type::addType(). You can get a list of all the known types with \Doctrine\DBAL\Types\Type::getTypesMap(). If this error occurs during database introspection then you might have forgot to register all database types for a Doctrine Type. Use AbstractPlatform#registerDoctrineTypeMapping() or have your custom types implement Type#getMappedDatabaseTypes(). If the type name is empty you might have a problem with the cache or forgot some mapping information.
```

with this code:

```
DB::getDoctrineSchemaManager()->getDatabasePlatform()->registerDoctrineTypeMapping('enum', 'string');
Schema::table('replays', function(Blueprint $table)
{
    $table->enum('game_type', ['QuickMatch', 'UnrankedDraft', 'HeroLeague', 'TeamLeague', 'Brawl', 'StormLeague'])->nullable()->index()->change();
});
```
[The documentation](https://laravel.com/docs/5.5/migrations) seems to indicate that enum columns can not be changed:

```
Only the following column types can be "changed": bigInteger, binary, boolean, date, dateTime, dateTimeTz, decimal, integer, json, longText, mediumText, smallInteger, string, text, time, unsignedBigInteger, unsignedInteger and unsignedSmallInteger.
```

Reply:
OK, merging

// IMO it is more readable to use double outer quotes to avoid escaping single ones